src/swsssdk/dbconnector.py

```diff
@@ -9,182 +9,324 @@
+                            logger.warning(msg)
+                            continue
+
+                        with open(db_include_file, "r") as inc_file:
+                            SonicDBConfig._sonic_db_config[namespace] = json.load(inc_file)
+            else:
+                sonic_db_file_path = SonicDBConfig.SONIC_DB_CONFIG_FILE
+                if os.path.isfile(sonic_db_file_path) ==  False:
+                    msg = "'{}' is not found, it is not expected in production devices!!".format(sonic_db_file_path)
+                    logger.warning(msg)
+                    sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
+
+                with open(sonic_db_file_path, "r") as read_file:
+                    SonicDBConfig._sonic_db_config['default'] = json.load(read_file)
+
         except (OSError, IOError):
             msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
             logger.exception(msg)
             raise RuntimeError(msg)
+
         SonicDBConfig._sonic_db_config_init = True
 
     @staticmethod
-    def db_name_validation(db_name):
+    def db_name_validation(db_name, ns_instance):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        if db_name not in SonicDBConfig._sonic_db_config["DATABASES"]:
+        data=SonicDBConfig._sonic_db_config[ns_instance]["DATABASES"]
+        if db_name not in data.keys():
             msg = "{} is not a valid database name in configuration file".format(db_name)
             logger.exception(msg)
             raise RuntimeError(msg)
 
     @staticmethod
-    def inst_name_validation(inst_name):
+    def inst_name_validation(inst_name, ns_instance):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        if inst_name not in SonicDBConfig._sonic_db_config["INSTANCES"]:
+        data = SonicDBConfig._sonic_db_config[ns_instance]["INSTANCES"]
+        if inst_name not in data.keys():
             msg = "{} is not a valid instance name in configuration file".format(inst_name)
             logger.exception(msg)
             raise RuntimeError(msg)
 
     @staticmethod
-    def get_dblist():
+    def get_dblist(ns_instance):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        return SonicDBConfig._sonic_db_config[ns_instance]["DATABASES"].keys()
+
+    @staticmethod
+    def is_instance_valid(ns_instance):
         if SonicDBConfig._sonic_db_config_init == False:
             SonicDBConfig.load_sonic_db_config()
-        return SonicDBConfig._sonic_db_config["DATABASES"].keys()
+        if SonicDBConfig._sonic_db_config[ns_instance] != None:
+            return True
+        else: 
+            return False
 
     @staticmethod
-    def get_instance(db_name):
+    def get_dbConnection_instances():
```

Comment:
>get_dbConnection_instances [](start = 8, length = 26)

get_namespace_list?

Reply:
Updated by changing instance to namespace at most places.